src/waveform/renderers/waveformbeat.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <proto/beats.pb.h>
+
+#include <QPainter>
+
+using BeatType = mixxx::track::io::Type;
+
+enum Direction {
+    UP = 0,
+    DOWN,
+    LEFT,
+    RIGHT
+};
+
+class WaveformBeat {
+  public:
+    WaveformBeat();
+    void draw(QPainter* painter) const;
+    void setPosition(int position) {
+        m_position = position;
+    }
+    void setType(BeatType beatType) {
```

Comment:
Do you really need to modify the type or orientation after construction? Try to make as many members immutable as possible. Dumb setters are considered an anti-pattern.